lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -176,17 +177,26 @@ protected function _getPortableTableColumnDefinition($tableColumn)
                 break;
         }
 
-        $length = ((int) $length == 0) ? null : (int) $length;
+        $length = ((int) $length) ?: null;
```

Comment:
While it's being changed, can we only cast `$length` to `int` when it's knowingly not an `int` (the very first few lines of the method)? Otherwise, the `$length` can be `NULL` (see a couple of lines above) which is cast to `int` and then is thrown away in favor of another `NULL`.